atomic_reactor/__init__.py

```diff
@@ -32,6 +35,28 @@ def format(self, record):
         return super(ArchFormatter, self).format(record)
 
 
+class EncodedStream(object):
+    def __init__(self, fileno, encoding):
+        self.binarystream = fdopen(dup(fileno), 'wb')
```

Comment:
Removing the file descriptor duplication bit causes this error: `ValueError: I/O operation on closed file`. That looks like a good reason to keep this. Thank you for adding the big fat warning comment.